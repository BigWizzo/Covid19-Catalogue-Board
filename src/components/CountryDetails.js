import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Pie } from 'react-chartjs-2';
import { useParams, Link } from 'react-router-dom';
import getCountryData from '../redux/actions/countryActions';
import CountryDescription from './CountryDescription';

const CountryDetails = () => {
  const countryParam = useParams();
  const dispatch = useDispatch();
  const country = useSelector((state) => state.country);
  const fetchCountry = () => {
    dispatch(
      getCountryData({
        nation: countryParam.country,
      }),
    );
  };

  useEffect(() => {
    fetchCountry();
  }, []);

  return (
    <div>
      <p className="text-success m-3">
        <Link className="text-success text-decoration-none bold" to="/">
          Go Home
        </Link>
      </p>
      <h1 className="text-center mt-3">{countryParam.country}</h1>
      {country.data ? (
        <div className="main-content-wrapper row m-0 p-3">
          <div className="chart-wrapper col-md-6">
            <Pie className="w-75 m-auto" data={country.data} />
          </div>
          <div className="country-desc col-md-6">
            <CountryDescription description={country.description} />
          </div>
        </div>
      ) : (
        <div className="empty text-center">
          {!country.loading && (
            <h4 className="no-entry p-3">
              Please enter Country Name and hit Submit
            </h4>
          )}
        </div>
      )}
    </div>
  );
};

export default CountryDetails;
