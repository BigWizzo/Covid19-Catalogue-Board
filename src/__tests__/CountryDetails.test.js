import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import { BrowserRouter } from 'react-router-dom';
import { store, persistor } from '../redux/store';
import CountryDetails from '../components/CountryDetails';

test('should render the country details page', () => {
  const component = renderer.create(
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <BrowserRouter>
          <CountryDetails />
        </BrowserRouter>
      </PersistGate>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
