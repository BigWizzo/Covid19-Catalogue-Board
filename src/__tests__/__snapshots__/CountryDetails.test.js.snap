// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the country details page 1`] = `
<div>
  <p
    className="text-success m-3"
  >
    <a
      className="text-success text-decoration-none bold"
      href="/"
      onClick={[Function]}
    >
      Go Home
    </a>
  </p>
  <h1
    className="text-center mt-3"
  />
  <div
    className="main-content-wrapper row m-0 p-3"
  >
    <div
      className="chart-wrapper col-md-6"
    >
      <canvas
        className="w-75 m-auto"
        data-testid="canvas"
        height={150}
        onClick={[Function]}
        role="img"
        width={300}
      />
    </div>
    <div
      className="country-desc col-md-6"
    >
      <div
        className="mb-3"
      >
        <div
          className="row g-0"
        >
          <div
            className="col-md-8"
          >
            <div
              className="card-body text-left"
            >
              <p
                className="country-wrap font-22"
              >
                <span>
                  Country Name: 
                </span>
                <span
                  className="country-code green-800 font-22"
                />
              </p>
              <p
                className="card-text"
              >
                <span>
                  Country Code: 
                </span>
                <span
                  className="card-title"
                />
              </p>
              <p
                className="mute-p"
              >
                <span
                  className="text-muted"
                >
                   
                  has a population of
                  . It has so far recorded
                   
                   
                  deaths.
                   
                  cases have been confirmed and
                   
                  people have recovered from the virus.
                </span>
              </p>
            </div>
          </div>
          <div
            className="col-md-4 text-center"
          >
            <h4
              className="card-hed font-italic"
            />
            <hr />
            <div
              className="card-body"
            >
              <div
                className="icon-div py-3"
              >
                <div
                  className="icon-span "
                >
                  <div
                    className=""
                  >
                    <p
                      className="font-10 text-success"
                    >
                      Last Updated
                    </p>
                    <p
                      className="increase"
                    >
                      <i
                        className="fas fa-pen-square font-22 text-success"
                      />
                    </p>
                    <p
                      className="font-12 text-success"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
